packages/turf-boolean-contains/index.js

```diff
@@ -0,0 +1,208 @@
+var lineOverlap = require('@turf/line-overlap');
+
+/**
+ * Contains returns true if the second geometry is completely contained by the first geometry.
+ * The contains predicate returns the exact opposite result of the within predicate.
+ *
+ * @name contains
+ * @param {Geometry|Feature<any>} feature1 source
+ * @param {Geometry|Feature<any>} feature2 target
+ * @returns {Boolean} true/false
+ * @example
+ * var along = turf.contains(line, point);
+ */
+module.exports = function (feature1, feature2) {
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (geom1.type) {
+    case 'Point':
+        switch (geom2.type) {
+        case 'Point':
+            return deepEqual(geom1.coordinates, geom2.coordinates);
+        }
+        throw new Error('feature2 ' + geom2.type + ' geometry not supported');
+    case 'MultiPoint':
+        switch (geom2.type) {
+        case 'Point':
+            return isPointInMultiPoint(geom1, geom2);
+        case 'MultiPoint':
+            return isMultiPointInMultiPoint(geom1, geom2);
+        }
+        throw new Error('feature2 ' + geom2.type + ' geometry not supported');
+    case 'LineString':
+        switch (geom2.type) {
+        case 'Point':
+            return isPointOnLine(geom1, geom2);
+        case 'LineString':
+            return isLineOnLine(geom1, geom2);
+        case 'MultiPoint':
+            return isMultiPointOnLine(geom1, geom2);
+        }
+        throw new Error('feature2 ' + geom2.type + ' geometry not supported');
+    case 'Polygon':
+        switch (geom2.type) {
+        case 'Point':
+            return isPointInPoly(geom1, geom2);
+        case 'LineString':
+            return isLineInPoly(geom1, geom2);
+        case 'Polygon':
+            return isPolyInPoly(geom1, geom2);
+        case 'MultiPoint':
+            return isMultiPointInPoly(geom1, geom2);
+        }
+        throw new Error('feature2 ' + geom2.type + ' geometry not supported');
+    default:
+        throw new Error('feature1 ' + geom1.type + ' geometry not supported');
+    }
+};
+
+function isPointInMultiPoint(MultiPoint, Point) {
+    var i;
+    var output = false;
+    for (i = 0; i < MultiPoint.coordinates.length; i++) {
+        if (deepEqual(MultiPoint.coordinates[i], Point.coordinates)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointInMultiPoint(MultiPoint1, MultiPoint2) {
+    var foundAMatch = 0;
+    for (var i = 0; i < MultiPoint2.coordinates.length; i++) {
+        var anyMatch = false;
+        for (var i2 = 0; i2 < MultiPoint1.coordinates.length; i2++) {
+            if (deepEqual(MultiPoint2.coordinates[i], MultiPoint1.coordinates[i2])) {
+                foundAMatch++;
+                anyMatch = true;
+                break;
+            }
+        }
+        if (!anyMatch) {
+            return false;
+        }
+    }
+    return foundAMatch > 0 && foundAMatch < MultiPoint1.coordinates.length;
```

Comment:
The second condition here excludes the two geometries being equal. Not sure if that is actually in the definition of `contains`. According to [DE-9IM](https://en.wikipedia.org/wiki/DE-9IM) the definition is:

 - No point of `MultiPoint2` must be in the exterior of `MultiPoint1`
 - At least 1 point in the interior of `MultiPoint2` must be in the interior of `MultiPoint1`

Since the points themselves are their own interior, the two being equal would seem to meet that criteria. Could be wrong, the DE-9IM definitions are sometimes tricky to parse.